README.md

```diff
@@ -0,0 +1,91 @@
+# dss-bridge
+
+Connect an instance of MCD to a cross-chain guest instance. The `DomainHost` is designed to plug into the Host's dss instance and the `DomainGuest` controls the Guest dss instance on the remote chain. For example, `DomainHost` could be plugged into the MCD master instance on Ethereum and a `DomainGuest` controller would be plugged into the Optimism slave instance of MCD. This is recursive, so that when layer 3s come online we can use another Host/Guest pairing on the L2.
+
+![dss-bridge](https://imgur.com/uEruNWB.png)
+
+Each contract is abstract as it requires a chain-specific messaging service, so one of these needs to be extended for each unique message interface.
+
+We assume any messaging bridge will guarantee messages will be executed exactly once, but does not enforce the ordering of messages. Messaging ordering for sensitive operations is enforced at the application layer (`DomainHost` or `DomainGuest`). If the messaging bridge does not protect against censorship then we cannot provide gaurantees that the bridge will function correctly.
+
+## Supported Operations
+
+### `DomainHost.deposit(address to, uint256 amount)`
+
+Standard deposit mechanism. Locks local DAI and mints canonical DAI on the guest domain.
+
+This will trigger a call to `DomainGuest.deposit(address to, uint256 amount)`.
+
+### `DomainGuest.withdraw(address to, uint256 amount)`
+
+Standard withdrawal mechanism. Burns local canonical DAI and unlocks the escrowed DAI on the host domain.
+
+This will trigger a call to `DomainHost.withdraw(address to, uint256 amount)`.
+
+### `DomainHost.lift(uint256 wad)`
+
+Specify a new global debt ceiling for the remote domain. If the amount is an increase this will pre-mint ERC20 and stick it in the domain escrow, so the future minted DAI on the remote domain is backed. It will not remove escrow DAI upon lowering as that can potentially lead to unbacked DAI. Releasing pre-minted DAI from the escrow needs to be handled by the remote domain.
+
+This will back the pre-minted DAI by `vat.gems` that represent shares to the remote domain.
+
+This will trigger a call to `DomainGuest.lift(uint256 _lid, uint256 rad)` with `rad == wad * RAY`.
+
+### `DomainGuest.surplus()`
+
+Permissionless function which will push a surplus to the host domain if the delta is greater than the dust limit. It will exit the DAI, send it across the token bridge leaving it ready for being realized by governance.
```

Comment:
```suggestion
Permissionless function which will push a surplus to the host domain if the delta is greater than the dust limit. It will exit the DAI, send it across the token bridge leaving it ready for being released by governance.
```

Reply:
Hmm actually here I was trying to say "realized" in the context of getting properly sent to the surplus buffer. Maybe is not a correct word for describing that, but don't think "released" is the one to replace it.